pydrumscore/export.py

```diff
@@ -528,6 +529,34 @@ def add_note(chord, notedef):
     with open(save_path, "wb") as f:
         f.write(xml_str)
 
+    zip_file= zipfile.ZipFile("pydrumscore/_exported/" + metadata.workTitle + ".mscz", "w")
+    ########################## Export from: export_files
+    # suggestion: Make a little function and make this locally
+    zip_file.write("pydrumscore/_exported/" + filename, filename, compress_type=zipfile.ZIP_DEFLATED)
+    zip_file.write("pydrumscore/export_files/audiosettings.json", "audiosettings.json", compress_type=zipfile.ZIP_DEFLATED)
+    zip_file.write("pydrumscore/export_files/score_style.mss", "score_style.mss", compress_type=zipfile.ZIP_DEFLATED)
+    zip_file.write("pydrumscore/export_files/viewsettings.json", "viewsettings.json", compress_type=zipfile.ZIP_DEFLATED)
+    zip_file.write("pydrumscore/export_files/Thumbnails/thumbnail.png", "Thumbnails/thumbnail.png", compress_type=zipfile.ZIP_DEFLATED)
+
+    ############################ Create container.xml
+    import xml.etree.ElementTree as ET
+    mytree = ET.parse("pydrumscore/export_files/META-INF/container.xml")
+    myroot= mytree.getroot()
+
+    # locate the tag you want to modify
+    for elem in myroot.iter():
+        if elem.tag == "rootfile" and "full-path" in elem.attrib:
+            # make the desired changes to the tag
+            elem.set("full-path", filename)
+            break
+
+
+    mytree.write("pydrumscore/export_files/META-INF/container.xml")
+    zip_file.write("pydrumscore/export_files/META-INF/container.xml", "META-INF/container.xml", compress_type=zipfile.ZIP_DEFLATED)
+
+    ###########################
+    zip_file.close()
```

Comment:
So when the zip file is created it is implicitly opened on line 532? Just curious since we have a `close` but no `open`.

Reply:
yes, the second argument to ZipFile is a single-letter string representing the mode that we’ll use to open the file.